### ISSUE TITLE
bug: range spec

### ISSUE THREAD

**Issue by ramfox** [n0-computer/iroh#1416]
Ended a transfer early after getting 199.67MiB of a 348.62MiB transfer, on the getter side.  I did not close the provider between requests. When starting the getter side back up, I got this bug on the provider:

```
thread '<unnamed>' panicked at 'attempt to subtract with overflow', iroh-bytes/src/protocol/range_spec.rs:179:12
```

**Comment by divagant-martian** [comment]
yeah I mentioned this here https://github.com/n0-computer/iroh/pull/1402#discussion_r1304562753

**Comment by rklaehn** [comment]
Is this fixed?

**Comment by divagant-martian** [comment]
closed by #1421
